applications/MultilevelMonteCarloLaplacianApplication/python_scripts/multilevel_montecarlo_laplacian_analysis.py

```diff
@@ -0,0 +1,724 @@
+
+
+    
+###########################################################
+######## END OF CLASS MONTECARLOLAPLACIANANALYSIS #########
+###########################################################
+
+
+
+def GenerateStochasticContribute(parameters):
+    number_samples = 1 # i.e. generate one stochastic variable per time
+
+    if (parameters["problem_data"].Has("stochastic_pdf")):
+        stochastic_pdf = parameters["problem_data"]["stochastic_pdf"]
+        print(stochastic_pdf)
+    else:
+        raise Exception('Please provide the "stochastic_pdf" parameter in the .json file')
+
+    if stochastic_pdf.Has("normal_distribution"):
+        if stochastic_pdf["normal_distribution"].Has("mean"):
+            mu = stochastic_pdf["normal_distribution"]["mean"].GetDouble()
+        else:
+            raise Exception('Please define the "mean" for the normal distribution in the .json file')
+        if stochastic_pdf["normal_distribution"].Has("variance"):
+            sigma = stochastic_pdf["normal_distribution"]["variance"].GetDouble()
+        else:
+            raise Exception('Please define the "variance" for the normal distribution in the .json file')
+        sample = np.random.normal(mu,sigma,number_samples)
+
+    elif stochastic_pdf.Has("beta_distribution"):
+        if stochastic_pdf["beta_distribution"].Has("alpha"):
+            alpha = stochastic_pdf["beta_distribution"]["alpha"].GetDouble()
+        else:
+            raise Exception('Please define the "alpha" for the beta distribution in the .json file')
+        if stochastic_pdf["beta_distribution"].Has("beta"):
+            beta = stochastic_pdf["beta_distribution"]["beta"].GetDouble()
+        else:
+            raise Exception('Please define the "beta" for the beta distribution in the .json file')
+        sample = np.random.beta(alpha,beta,number_samples)
+
+    else:
+        raise Exception('Please provide "normal_distribution" or "beta_distribution" in the .json file; at the moment only "normal_distribution" and "beta_distribution" are implemented')
+    return sample
+
+def GenerateBetaSample(alpha,beta):
+    number_samples = 1
+    sample = np.random.beta(alpha,beta,number_samples)
+    return sample
+
+
+def EvaluateQuantityOfInterest(simulation):
+    """here we evaluate the QoI of the problem: int_{domain} SOLUTION(x,y) dx dy
+    we use the midpoint rule to evaluate the integral"""
+    Kratos.CalculateNodalAreaProcess(simulation._GetSolver().main_model_part,2).Execute()
+    Q = 0.0
+    for node in simulation._GetSolver().main_model_part.Nodes:
+        Q = Q + (node.GetSolutionStepValue(Kratos.NODAL_AREA)*node.GetSolutionStepValue(Poisson.SOLUTION))
+        #print("NODAL AREA = ",node.GetSolutionStepValue(Kratos.NODAL_AREA),"NODAL SOLUTION = ",node.GetSolutionStepValue(Poisson.SOLUTION),"CURRENT Q = ",Q)
+    return Q
+
+
+def Nf_law(lev):
```

Comment:
what is this function doing? comment it ... and ideally use python multilne comment '''text'''